overlord/servicestate/quota_control_test.go

```diff
@@ -111,28 +159,53 @@ func checkQuotaState(c *C, st *state.State, exp map[string]quotaGroupState) {
 	}
 }
 
-func checkSvcAndSliceState(c *C, snapSvc string, slicePath string, sliceMem quantity.Size) {
+// shouldMentionSlice returns whether or not a slice file
+// should be mentioned in the service unit file. It does in the case
+// when a quota is set.
+func shouldMentionSlice(resources quota.Resources) bool {
+	if resources.Memory == nil && resources.CPU == nil &&
```

Comment:
I'd also remove this condition (and the function itself), and instead have the caller of `checkSvcAndSliceState()` specify whether the slice should be mentioned or not (that is, `checkSvcAndSliceState()` should take one more boolean parameter).

The point is that tests should have as little logic as possible, or we risk introducing errors here too. The dumber the test, the better :-)

Reply:
Maybe a change for a followup PR? I don't think that redoing how we test quotas should be in scope for this, as we are talking about 41 occurrences in atleast 20 tests.

Reply:
Oops, sure, and by the way we can do it little by little, no need to change all tests at once (unless it's really impossible).